elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchSourceStage.scala

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.elasticsearch
+
+import java.io.ByteArrayOutputStream
+import java.util
+
+import akka.stream.{Attributes, Outlet, SourceShape}
+import akka.stream.stage.{GraphStage, GraphStageLogic, OutHandler}
+import org.apache.http.entity.StringEntity
+import org.elasticsearch.client.{Response, RestClient}
+import spray.json._
+import DefaultJsonProtocol._
+
+import scala.collection.JavaConverters._
+
+final case class ElasticsearchSourceSettings(bufferSize: Int = 10)
+
+final case class OutgoingMessage[T](id: String, source: T)
+
+final class ElasticsearchSourceStage(indexName: String,
```

Comment:
Same as above, can probably be provided by creating a `JsonReader[JsObject]` and then using the typed variant.